### PR TITLE
Add a function to get a commit information from its SHA

### PR DESCRIPTION
**Pull request by vitaliil-jfrog**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,8 +24,9 @@ Currently supported providers are: [GitHub](#github), [Bitbucket Server](#bitbuc
   - [Delete Webhook](#delete-webhook)
   - [Set Commit Status](#set-commit-status)
   - [Create Pull Request](#create-pull-request)
-  - [Latest Commit Hash](#get-latest-commit-hash)
+  - [Latest Commit](#get-latest-commit)
```
Comment (yahavi) [inline README.md:27]: Let's put these two together, as there is a logical connection between them. Let's do it also on their description.
```suggestion
  - [Latest Commit](#get-latest-commit)
  - [Get Commit By SHA](#get-commit-by-sha)
```

File: vcsclient/github_test.go [hunk 2]
```diff
@@ -217,6 +217,48 @@ func TestGitHubClient_AddSshKeyToRepositoryReadWrite(t *testing.T) {
...
+		ParentHashes:  []string{"5dcb09b5b57875f334f61aebed695e2e4193db5e"},
+	}, result)
+}
+
+func TestGitHubClient_GetCommitByWrongSha(t *testing.T) {
+	ctx := context.Background()
+	sha := "5dcb09b5b57875f334f61aebed695e2e4193db5e"
+	response := []byte(`{
+    "message": "No commit found for SHA: 5dcb09b5b57875f334f61aebed695e2e4193db5e"
+}`)
```
Comment (yahavi) [inline vcsclient/github_test.go:249]:
```suggestion
	response := []byte(`{
		"message": "No commit found for SHA: 5dcb09b5b57875f334f61aebed695e2e4193db5e"
	}`)
```

File: vcsclient/gitlab_test.go [hunk 3]
```diff
@@ -234,6 +234,50 @@ func TestGitLabClient_AddSshKeyToRepositoryReadWrite(t *testing.T) {
...
+		ParentHashes:  []string{"667fb1d7f3854da3ee036ba3ad711c87c8b37fbd"},
+	}, result)
+}
+
+func TestGitLabClient_GetCommitByShaNotFound(t *testing.T) {
+	ctx := context.Background()
+	sha := "ff4a54b88fbd387ac4d9e8cdeb54b049978e450b"
+	response := []byte(`{
+    "message": "404 Commit Not Found"
+}`)
```
Comment (yahavi) [inline vcsclient/gitlab_test.go:267]:
```suggestion
	response := []byte(`{
		"message": "404 Commit Not Found"
	}`)
```

**Review by yahavi — CHANGES_REQUESTED** [review]
@vitaliil-jfrog I really liked the code!
Please see my inline comments.

File: README.md [hunk 4]
```diff
@@ -24,7 +24,8 @@ Currently supported providers are: [GitHub](#github), [Bitbucket Server](#bitbuc
...
```
Comment (ttetzlaff) [inline README.md:27]:
```suggestion
  - [Get Latest Commit](#get-latest-commit)
```

**Review by ttetzlaff — APPROVED** [review]

**Pull request merged**